sycl/doc/extensions/template.rst

```diff
@@ -0,0 +1,653 @@
+
+Ensure good HTML rendering
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Specifications should be written in reStructuredText, and we use Sphinx tools to
+generate HTML pages from the reStructuredText source code.
+However, we also expect many users to read the specifications directly from
+GitHub.
+If you open a file with the ``.rst`` suffix from GitHub in a browser, the GitHub
+server displays an HTML rendering of the file.
+Therefore, there are two ways these specifications can be rendered into HTML:
+via the Sphinx tools and via the GitHub server.
+We expect these specifications to be nicely formatted in both cases.
+
+When writing a specification, make sure your document is rendered nicely in both
+cases.
+You can generate the HTML locally via Sphinx with the following commands:
+
+.. code-block:: bash
+
+   $ cd build
+   $ cmake --build . --target docs-sycl-html
+
+And the HTML will be generated in ``build/tools/sycl/doc/html/extensions``.
+
+To check the HTML rendering via GitHub, push your changes to a branch in
+GitHub, and then use a browser to navigate to your specification file(s).
+
+Describe your extension not how to change the SYCL specification
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Do not describe **how** you will change the core SYCL specification.
+For example, there is no need to say something like "Add the following paragraph
+to section 4.6.7 of the core SYCL specification ..."
+Instead, just describe the semantics and APIs of the extension itself.
+We will figure out later how to change the core SYCL specification if the
+extension is adopted.
+
+Avoid unnecessary duplication of the SYCL specification
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Avoid unnecessary duplication of APIs or text from the core SYCL specification.
+For example, if your extension adds a new member function to an existing class,
+there is no need to duplicate the existing member functions in your
+specification.
+Duplication like this can be troublesome if something in the core SYCL
+specification changes later.
+
+Use ISO C++ "words of power"
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When specifying a function, use paragraphs with the following titles:
+
+* *Constraints:* This paragraph can be used whenever the function is templated
+  or when the function is a member of a templated class.
+  The constraints describe the SFINAE constraints or ``requires`` constraints
+  of the template parameters.
+
+* *Preconditions:* This paragraph tell the conditions that the application must
+  obey when calling the function.
+  If the application violates a precondition, the behavior is undefined.
+
+* *Effects:* Tells what the function does.
+
+* *Returns:* Tells what value the function returns.
+
+* *Throws:* Tells what exceptions the function is required to throw and the
+  circumstances under which they are thrown.
+
+* *Remarks:* Tells other information about the function if it is not covered by
+  the previous paragraphs.
```

Comment:
Nit: Something about "tells" feels off to me.  I'd prefer a word like "Lists" but I can't really explain why.